typescript/sdk/src/router/adapters/EvmRouterAdapter.ts

```diff
@@ -7,76 +7,68 @@ import {
 import { Address, Domain, bytes32ToAddress } from '@hyperlane-xyz/utils';
 
 import { BaseEvmAdapter } from '../../app/MultiProtocolApp';
+import { MultiProtocolProvider } from '../../providers/MultiProtocolProvider';
 import { ChainName } from '../../types';
-import { RouterAddress } from '../types';
 
 import { IGasRouterAdapter, IRouterAdapter } from './types';
 
-export class EvmRouterAdapter<
-    ContractAddrs extends RouterAddress = RouterAddress,
-  >
-  extends BaseEvmAdapter<ContractAddrs>
-  implements IRouterAdapter<ContractAddrs>
-{
-  interchainSecurityModule(chain: ChainName): Promise<Address> {
-    return this.getConnectedContract(chain).interchainSecurityModule();
+export class EvmRouterAdapter extends BaseEvmAdapter implements IRouterAdapter {
+  constructor(
+    public readonly chainName: ChainName,
+    public readonly multiProvider: MultiProtocolProvider<any>,
+    public readonly addresses: { router: Address },
+  ) {
+    super(chainName, multiProvider);
+  }
+
+  interchainSecurityModule(): Promise<Address> {
```

Comment:
how useful is this if we are just passing through to the contract?

Reply:
The idea is that adapter consumers, which may be MultiProtocolApps or just devs using them directly, don't need to know the details about the contract/program. This is trivial in the EVM case because EVM is reasonable. It involves a bunch of program-specific deserialization in the Sealevel case though